### PR TITLE
Added better Typescript support

### PR DESCRIPTION
**Pull request by tomitrescak**
Hi, this pull request adds the same level of typescript support as tsx-control-statements. Sorry for the READMe.md, looks like my VSCODe reformatted it to its own format, if you wish I can undo those changes and try with the original format.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ module.exports = function jcsPlugin(babel) {
 
   var visitor = {
     JSXElement: function(path) {
+      // console.log(path)
```
Comment (AlexGilleran) [inline src/index.js]: Would be good to remove this :)

**Comment by tomitrescak** [comment]
Hmmm, that makes sense. Yes, I guess you do not need to transform For, I did not even think of that. 

Let me know if you still want to merge and if yes, I can fix linting errors.

**Comment by AlexGilleran** [comment]
Yeah I'm still happy to merge - to be honest all of jsx-control-statements would probably be better done with render props, so I don't think it's unreasonable to have it in here, was just curious :).

**Comment by tomitrescak** [comment]
Hi, I've been busy .. what can I do for you to merge this? We are finally deploying the project and need this published for CI/CD purposes.

**Comment by AlexGilleran** [comment]
Just needs the lint problems that are breaking travis resolved :)

**Comment by tomitrescak** [comment]
Ok, should be fixed now. Thanks!

**Comment by tomitrescak** [comment]
@AlexGilleran do you plan to release this? I just need to know if I should update my package.jsons to install from github

**Comment by AlexGilleran** [comment]
Working on it :). I'm trying to upgrade all the deps, which is causing some
test failures, but I'm nearly through them.

**Comment by AlexGilleran** [comment]
All done, you can give it a try as 4.1.0!

**Comment by tomitrescak** [comment]
Awesome! Thanks

**Pull request merged**